src/pipelines/sentence_embeddings/pipeline.rs

```diff
@@ -0,0 +1,370 @@
+                .as_deref(),
+            tokenizer_config.do_lower_case,
+            strip_accents,
+            add_prefix_space,
+        )?;
+
+        // Setup pooling layer
+
+        let pooling_config = PoolingConfig::from_file(pooling_config_resource.get_local_path()?);
+        let pooling_layer = Pooling::new(pooling_config);
+
+        // Setup dense layer
+
+        let mut dense_layer = None;
+        if modules.dense_module().is_some() {
+            let dense_config =
+                DenseConfig::from_file(dense_config_resource.unwrap().get_local_path()?);
+            dense_layer = Some(Dense::new(
+                dense_config,
+                dense_weights_resource.unwrap().get_local_path()?,
+                device,
+            )?);
+        }
+
+        let normalize_embeddings = modules.has_normalization();
+
+        Ok(Self {
+            tokenizer,
+            tokenizer_config,
+            tokenizer_truncation_strategy: TruncationStrategy::LongestFirst,
+            var_store,
+            transformer,
+            transformer_config,
+            pooling_layer,
+            dense_layer,
+            normalize_embeddings,
+        })
+    }
+
+    pub fn set_tokenizer_truncation(&mut self, truncation_strategy: TruncationStrategy) {
+        self.tokenizer_truncation_strategy = truncation_strategy;
+    }
+
+    pub fn tokenize<S>(&self, inputs: &[S]) -> SentenceEmbeddingsTokenizerOuput
+    where
+        S: AsRef<str> + Sync,
+    {
+        let tokenized_input = self.tokenizer.encode_list(
+            inputs,
+            self.tokenizer_config.max_seq_length,
+            &self.tokenizer_truncation_strategy,
+            0,
+        );
+
+        let max_len = tokenized_input
+            .iter()
+            .map(|input| input.token_ids.len())
+            .max()
+            .unwrap_or(0);
+
+        let pad_token_id = self.tokenizer.get_pad_id().unwrap_or(0);
+        let tokens_ids = tokenized_input
+            .into_iter()
+            .map(|input| {
+                let mut token_ids = input.token_ids;
+                token_ids.extend(vec![pad_token_id; max_len - token_ids.len()]);
+                token_ids
+            })
+            .collect::<Vec<_>>();
+
+        let tokens_masks = tokens_ids
+            .iter()
+            .map(|input| {
+                Tensor::of_slice(
+                    &input
+                        .iter()
+                        .map(|&e| if e == pad_token_id { 0_i64 } else { 1_i64 })
+                        .collect::<Vec<_>>(),
+                )
+            })
+            .collect::<Vec<_>>();
+
+        let tokens_ids = tokens_ids
+            .into_iter()
+            .map(|input| Tensor::of_slice(&(input)))
+            .collect::<Vec<_>>();
+
+        SentenceEmbeddingsTokenizerOuput {
+            tokens_ids,
+            tokens_masks,
+        }
+    }
+
+    pub fn forward<S>(&self, inputs: &[S]) -> Result<SentenceEmbeddingsModelOuput, RustBertError>
```

Comment:
I feel it's interesting to be able to have it in case you want the result as a `Tensor` (all `forward(...)` methods return `Tensor`s) and let `encode(...)` be specialized by returning a `Vec<Embedding>`.

Reply:
I understand - can we instead make `forward` private, and create a new `encode_as_tensors` method that would return the tensor output? My concern is the difference between forward and encode not being clear, and I believe it would be easier if all methods would be called `encode_[...]` describing what they return.

Reply:
Could you please also make sure that all public structures and methods have a docstring, taking as a model the other pipelines?

Reply:
> I understand - can we instead make `forward` private, and create a new `encode_as_tensors` method that would return the tensor output? My concern is the difference between forward and encode not being clear, and I believe it would be easier if all methods would be called `encode_[...]` describing what they return.

Done